provider.tf

```diff
@@ -2,7 +2,7 @@ terraform {
   required_providers {
     azuread = {
       source  = "hashicorp/azuread"
-      version = ">= 1.6.0"
+      version = ">= 2.2.1"
```

Comment:
does it need to be so high? we shouldn't force people to update for the sake of it.

isn't 2.0.0 enough?

remember this is a minimum version, not necessarily the latest